### ISSUE TITLE
Crash on enumerate with binary operator in start parameter

### ISSUE THREAD

**Issue by targodan** [PyCQA/pylint#7963]
### Bug description

Minimal example (in our code which initially triggered the bug, the addition consisted of one variable and one literal):

```python
def this_crashes_pylint(arr):
    for i, v in enumerate(arr, 1+1):
        pass

def this_is_fine(arr):
    for i, v in enumerate(arr, 1):
        pass
```

### Configuration

```ini
# Default pylint config i.e. no config.
```

### Command used

```shell
pylint crash.py
```

### Pylint output

```shell
************* Module crash
crash.py:8:0: C0305: Trailing newlines (trailing-newlines)
crash.py:1:0: C0114: Missing module docstring (missing-module-docstring)
crash.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
Exception on node <For l.2 at 0x7fccd1fa95d0> in file '/tmp/pylintcrash/crash.py'
Traceback (most recent call last):
  File "/tmp/pylintcrash/venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/tmp/pylintcrash/venv/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 682, in visit_for
    self._check_unnecessary_list_index_lookup(node)
  File "/tmp/pylintcrash/venv/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2118, in _check_unnecessary_list_index_lookup
    has_start_arg, confidence = self._enumerate_with_start(node)
  File "/tmp/pylintcrash/venv/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2236, in _enumerate_with_start
    start_val, confidence = self._get_start_value(start_arg)
  File "/tmp/pylintcrash/venv/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2260, in _get_start_value
    start_val = node.value
AttributeError: 'BinOp' object has no attribute 'value'
crash.py:1:0: F0002: crash.py: Fatal error while checking 'crash.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/corbatto/.cache/pylint/pylint-crash-2022-12-20-08-35-59.txt'. (astroid-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

Should not crash.

### Pylint version

```shell
pylint 2.15.9
astroid 2.12.13
Python 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0]

Likely introduced in this release, since we've been checking the exact same lines of code for months if not years without a crash.
```

### OS / Environment

ArchLinux with zen kernel 6.0.12. Terminal: alacritty + tmux + zsh. It also happened on an ubuntu based bamboo worker, don't know the exact version.

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue !

**Comment by clavedeluna** [comment]
This issue has already been fixed and does not happen on the current `main` branch. @Pierre-Sassoulas surprised the fix hasn't been released?  It was fixed here https://github.com/PyCQA/pylint/pull/7824 and should've been backported.

**Comment by Pierre-Sassoulas** [comment]
It broke the maintenance branch so we reverted in https://github.com/PyCQA/pylint/pull/7855. We could release 2.16.0, but it's blocked by astroid's 2.13.0 release, which itself is blocked by https://github.com/PyCQA/astroid/pull/1189 and https://github.com/PyCQA/astroid/pull/1910. Another way to fix this would be to do a proper cherry-pick on the maintenance branch.

**Comment by Pierre-Sassoulas** [comment]
We released astroid 2.13.0, once #8030 is merged and once the blocker issues are resolved we're going to be able to release pylint 2.16.0.
